--- shibboleth-sp-3.0.4/configs/native.logger.org	2025-03-07 06:08:36.758263119 +0000
+++ shibboleth-sp-3.0.4/configs/native.logger	2025-03-07 06:11:27.954263119 +0000
@@ -27,12 +27,9 @@ log4j.category.XMLTooling.libcurl=WARN
 
 # define the appender
 
-# Change to SyslogAppender for remote syslog, and set host/port
-log4j.appender.native_log=org.apache.log4j.LocalSyslogAppender
-#log4j.appender.native_log.syslogHost=localhost
-#log4j.appender.native_log.portNumber=514
-log4j.appender.native_log.syslogName=shibboleth
-# Facility is numeric, 16 is LOCAL0
-log4j.appender.native_log.facility=16
+log4j.appender.native_log=org.apache.log4j.RollingFileAppender
+log4j.appender.native_log.fileName=/var/log/shibboleth/native.log
+log4j.appender.native_log.maxFileSize=1000000
+log4j.appender.native_log.maxBackupIndex=10
 log4j.appender.native_log.layout=org.apache.log4j.PatternLayout
-log4j.appender.native_log.layout.ConversionPattern=%p %c %x: %m%n
+log4j.appender.native_log.layout.ConversionPattern=%d{%Y-%m-%d %H:%M:%S} %p %c %x: %m%n
